File: frontend-redesign1/types/analysis.ts
export interface Analysis {
  id: number;
  job_id: string;
  error_type: string;
  root_cause: string;
  explanation: string;
  fix_suggestion: string;
  fix_command: string;
  analysis_source: string;
  created_at: string;
}

export type JobStatusValue =
  | "PENDING"
  | "RUNNING"
  | "COMPLETED"
  | "FAILED"
  | string;

export interface Job {
  job_id: string;
  status: JobStatusValue;
  current_step: string;
  analysis?: Analysis | null;
  error?: string | null;
}

File: backend/app/models/analysis.py
from sqlalchemy import Column, Integer, Text, String, DateTime
from datetime import datetime

from app.database.base import Base


class Analysis(Base):
    __tablename__ = "analyses"

    id = Column(Integer, primary_key=True, index=True)

    original_log = Column(Text, nullable=False)
    cleaned_log = Column(Text, nullable=True)

    error_type = Column(String, nullable=True)
    root_cause = Column(Text, nullable=True)

    explanation = Column(Text, nullable=True)

    fix_suggestion = Column(Text, nullable=True)
    fix_command = Column(Text, nullable=True)
    exit_code = Column(Integer, nullable=True)
    analysis_source = Column(
    String,
    nullable=False,
    )
    created_at = Column(
        DateTime,
        default=datetime.utcnow
        
    )
    job_id = Column(
         String,
         unique=True,
         nullable=False,
    )